Email opt-in for Email Newsletter on portal

riaJha97 opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/702)
<!-- Reviewable:end -->

OlafSzmidt commented:
Review status: 0 of 13 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

*[portal/urls.py, line 165 at r1](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-L9ybFHJ6AA2kGaY5sec:-L9ybFHKCkXFw-G3cDe2:bwanrvy) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/6d868e4b24edca8a6409eca49a66d0160ef050b3/portal/urls.py#L165)):*
> ```Python
>     ])),
> 
>     url(r'^news_signup', process_newsletter_form, name='process_newsletter_form'),
> ```

Can we move this up closer to  `verify_email` and those views there? Not important but more relevant up there

---

*[portal/helpers/emails.py, line 128 at r1](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-L9ycYPz2KLBD50e2nVl:-L9ycYPz2KLBD50e2nVm:b-q0z5ec) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/6d868e4b24edca8a6409eca49a66d0160ef050b3/portal/helpers/emails.py#L128)):*
> ```Python
> 
> 
> def add_to_salesforce(first_name, last_name, email):
> ```

I'm pretty sure this will break the change that was done in #696 ? It uses a django auth user definition for this method.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702)*
<!-- Sent from Reviewable.io -->

OlafSzmidt commented:
Review status: 0 of 13 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[portal/static/portal/sass/partials/_forms.scss, line 217 at r1](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-L9yd4H2DM6g0dqCknK9:-L9yd4H2DM6g0dqCknKA:b-o7ybkc) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/6d868e4b24edca8a6409eca49a66d0160ef050b3/portal/static/portal/sass/partials/_forms.scss#L217)):*
> ```SCSS
> 
> .form--news-signup {
>   //width: 700px;
> ```

Commented code

---

*[portal/static/portal/sass/partials/_forms.scss, line 254 at r1](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-L9yd8EgDJHg74xSaeiR:-L9yd8EgDJHg74xSaeiS:b-o7ybkc) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/6d868e4b24edca8a6409eca49a66d0160ef050b3/portal/static/portal/sass/partials/_forms.scss#L254)):*
> ```SCSS
>   //top: -8px;
>   //margin-right: -50px;
>   //display: inline-block;
> ```

Commented code

---

*[portal/templates/portal/base.html, line 238 at r1](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-L9ydCZ5BM_qFKxSSWBu:-L9ydCZ5BM_qFKxSSWBv:b-yiqp6s) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/6d868e4b24edca8a6409eca49a66d0160ef050b3/portal/templates/portal/base.html#L238)):*
> ```HTML
>             {% if messages %}
>                 <ul class="messages">
> ```

Can we remove this blank line?

---

*[portal/views/home.py, line 403 at r1](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-L9ydKQz3UZdD_IPytaP:-L9ydKQz3UZdD_IPytaQ:b-ssgl2z) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/6d868e4b24edca8a6409eca49a66d0160ef050b3/portal/views/home.py#L403)):*
> ```Python
> 
>     if request.method == 'POST':
> ```

Rest of the repo doesn't have blank lines after conditionals 

---

*[portal/views/home.py, line 419 at r1](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-L9ydQxY5yAyzIf0q8b0:-L9ydQxY5yAyzIf0q8b1:blt148t) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/6d868e4b24edca8a6409eca49a66d0160ef050b3/portal/views/home.py#L419)):*
> ```Python
> 
> def home(request):
>     return render(request, 'portal/home.html')
> ```

EOF line

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702)*
<!-- Sent from Reviewable.io -->

OlafSzmidt commented:
Review status: 0 of 13 files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[portal/helpers/emails.py, line 128 at r1](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-L9ycYPz2KLBD50e2nVl:-L9ydWCM-iIO8Fx3lzQh:bauw54t) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/6d868e4b24edca8a6409eca49a66d0160ef050b3/portal/helpers/emails.py#L128)):*
<details><summary><i>Previously, OlafSzmidt (Olaf Szmidt) wrote…</i></summary><blockquote>

I'm pretty sure this will break the change that was done in #696 ? It uses a django auth user definition for this method.
</blockquote></details>

Scrap that, just seen your change

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702)*
<!-- Sent from Reviewable.io -->

OlafSzmidt commented:
Reviewed 13 of 13 files at r1.
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702)*
<!-- Sent from Reviewable.io -->

j4mesholland commented:
Invalid email address warning isn't rendering correctly and is the wrong kind of alert.

![screenshot from 2018-04-13 12-04-53](https://user-images.githubusercontent.com/18662916/38731623-06fe4b28-3f13-11e8-8e0e-60f13bb31835.png)

The alert should be a waning rather than just info. It should look something like the yellow alert:
![screen shot 2018-04-13 at 12 09 00](https://user-images.githubusercontent.com/18662916/38731721-7127d47e-3f13-11e8-8c1e-49c595d28117.png)

CelineBoudier commented:
Reviewed 2 of 13 files at r1.
Review status: 9 of 13 files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[portal/context_processors.py, line 1 at r2](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-L9z27ob3V_JaWNbPhtk:-L9z27ob3V_JaWNbPhtl:b-wpq72h) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/f79ad78b342044a6ed247a8a9cf922ec45157ec5/portal/context_processors.py#L1)):*
> ```Python
> from portal.forms.newsletter_form import NewsletterForm
> ```

if it's a new file we might want to add the copyright notice here as well

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702)*
<!-- Sent from Reviewable.io -->

CelineBoudier commented:
Reviewed 7 of 13 files at r1, 4 of 4 files at r2.
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[portal/static/portal/sass/partials/_forms.scss, line 208 at r2](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-L9zOUga9zpJcOg600fz:-L9zOUgb-ra60Uq7S-0r:bn1qqpg) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/b2a46607e92948422da38de3f6db174094703503/portal/static/portal/sass/partials/_forms.scss#L208)):*
> ```SCSS
> 
>     input,
>     button,
> ```

why has this been removed?

---

*[portal/tests/pageObjects/portal/home_page.py, line 58 at r2](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-L9zOhqp4TPUgg-TPDIT:-L9zOhqp4TPUgg-TPDIU:buxk0i2) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/f79ad78b342044a6ed247a8a9cf922ec45157ec5/portal/tests/pageObjects/portal/home_page.py#L58)):*
> ```Python
> 
>     def newsletter_singup(self, email):
>         self.browser.find_element_by_id('id_newsletter_email').send_keys(email)
> ```

as said earlier, this will not work for your element 

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702)*
<!-- Sent from Reviewable.io -->

OlafSzmidt commented:
Reviewed 4 of 4 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702)*
<!-- Sent from Reviewable.io -->

OlafSzmidt commented:
Review status: 8 of 13 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[portal/templates/portal/base.html, line 230 at r3](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LADS_Tc7_iWrwwk8TFV:-LADS_Tc7_iWrwwk8TFW:bxnecpj) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/34619bdeeff2c0dbc8117e1249fa46f2d626e0e8/portal/templates/portal/base.html#L230)):*
> ```HTML
>                 {% csrf_token %}
>                 <div class="form--news-signup">
>                     {{ news_form}}
> ```

Space between var and curly brackets

---

*[portal/views/home.py, line 410 at r3](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LADSdzyDuCm4j7D0SvH:-LADSdzyDuCm4j7D0SvI:bw6v59s) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/34619bdeeff2c0dbc8117e1249fa46f2d626e0e8/portal/views/home.py#L410)):*
> ```Python
>             return HttpResponseRedirect(request.META.get('HTTP_REFERER', '/'))
> 
>         messages.warning(request, 'Invalid email address. Please try again.', extra_tags='sub-nav sub-nav--warning')
> ```

Should this not be a `messages.error` call?

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702)*
<!-- Sent from Reviewable.io -->

CelineBoudier commented:
Reviewed 2 of 5 files at r3, 4 of 5 files at r4.
Review status: 13 of 14 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[portal/views/home.py, line 410 at r3](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LADSdzyDuCm4j7D0SvH:-LADqR6f4Q09hGAQ7DLm:bf3cpup) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/34619bdeeff2c0dbc8117e1249fa46f2d626e0e8/portal/views/home.py#L410)):*
<details><summary><i>Previously, OlafSzmidt (Olaf Szmidt) wrote…</i></summary><blockquote>

Should this not be a `messages.error` call?
</blockquote></details>

+1 - if message.error gives a warning instead, then it's an issue we need to fix

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702)*
<!-- Sent from Reviewable.io -->

CelineBoudier commented:
Reviewed 1 of 5 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[a discussion](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LAEkcNN7IixVHvf0yze:-LAEkcNN7IixVHvf0yzf:b-cughea) (no related file):*
"NoSuchElementException: Message: no such element: Unable to locate element: {"method":"id","selector":"messages"}" as seen in the Travis build

---

*[portal/tests/test_newsletter_footer.py, line 14 at r5](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LAEkoPK8cfxhhhqh2uz:-LAEkoPK8cfxhhhqh2v-:bd8z6fj) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/0e5a30402a4402155ae740f66122533be1132157/portal/tests/test_newsletter_footer.py#L14)):*
> ```Python
>         page.newsletter_singup(valid_email)
>         print "Sent valid post req"
>         print page.browser.page_source.encode("utf-8")
> ```

prints like this should not be committed (useful for testing locally i guess)

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702)*
<!-- Sent from Reviewable.io -->

CelineBoudier commented:
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[portal/views/home.py, line 92 at r5](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LAIDon-EOFefDqAt0of:-LAIDon-EOFefDqAt0og:bjw8coj) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/0e5a30402a4402155ae740f66122533be1132157/portal/views/home.py#L92)):*
> ```Python
> @ratelimit('email', labeller=teach_email_labeller, ip=False, periods=['1m'], increment=lambda req, res: hasattr(res, 'count') and res.count)
> @ratelimit('name', labeller=play_name_labeller, ip=False, periods=['1m'], increment=lambda req, res: hasattr(res, 'count') and res.count)
> def  render_login_form(request):
> ```

there are two spaces between def and render_login_form    There should be only one

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702)*
<!-- Sent from Reviewable.io -->

OlafSzmidt commented:
Review status: 6 of 14 files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[portal/tests/test_newsletter_footer.py, line 48 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LAJ1z_mAV0lI6VpW4FX:-LAJ1z_mAV0lI6VpW4FY:bjaimqj) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/tests/test_newsletter_footer.py#L48)):*
> ```
> resp
> ```

Let's call this response instead.

---

*[portal/tests/test_newsletter_footer.py, line 56 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LAJ21aqCPeoT5xIPCdn:-LAJ21aqCPeoT5xIPCdo:b-f7rbz9) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/tests/test_newsletter_footer.py#L56)):*
> ```
> resp
> ```

Same here.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702)*
<!-- Sent from Reviewable.io -->

OlafSzmidt commented:
Reviewed 1 of 5 files at r4, 8 of 8 files at r6.
Review status: all files reviewed at latest revision, 8 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702)*
<!-- Sent from Reviewable.io -->

mrniket commented on portal/context_processors.py:
```diff
@@ -0,0 +1,41 @@
+
+def process_newsletter_form(request):
+    return {'news_form': NewsletterForm()
+            }
```
new line here

mrniket commented on portal/forms/newsletter_form.py:
```diff
@@ -0,0 +1,43 @@
+class NewsletterForm(forms.Form):
+
+    email = forms.EmailField(label='Sign up to our newsletter', label_suffix="",
+                             widget=forms.TextInput(attrs={'placeholder': 'your.name@yourdomain.com', 'id': 'newsletter_email_field'}))
```
new line here too (Sorry)

mrniket commented on portal/static/portal/sass/partials/_footer.scss:
```diff
@@ -48,6 +48,11 @@
   .social-media a,
   .social-media a:hover {
     border-bottom: none;
+
```
there's an extra line here

mrniket commented on portal/static/portal/sass/partials/_forms.scss:
```diff
@@ -214,3 +212,26 @@ form {
+.form--news-signup > * {
+  margin-left: 10px;
+  margin-right: 10px;
+
```
extra line here

mrniket commented on portal/static/portal/sass/partials/_forms.scss:
```diff
@@ -214,3 +212,26 @@ form {
+  color: #ffffff;
+  white-space: nowrap;
+  margin-top: 20px;
+
```
extra line here

mrniket commented on portal/static/portal/sass/partials/_forms.scss:
```diff
@@ -214,3 +212,26 @@ form {
+.form--news-signup button {
+  margin-top: 10px;
+  white-space: nowrap;
+}
```
new line here

mrniket commented on portal/templates/portal/base.html:
```diff
@@ -168,13 +168,16 @@
             {% for message in messages %}
             <div class="sub-nav {{ message.tags }}">
+                {% if 'warning' in message.tags %}
+                <img title="Warning" alt="Warning sign" src="{% static 'portal/img/warning.png' %}">
```
does this get used if the message tag is error?

OlafSzmidt replied:
+1

mrniket commented on portal/templates/portal/base.html:
```diff
@@ -217,12 +222,24 @@
+                    {% csrf_token %}
+                    <label>{{ news_form.email.label }}</label>
+                    {{ news_form.email }}
+                    {{ news_form.non_field_errors }}
```
It might be worth asking @j4mesholland where this should go

mrniket commented on portal/tests/pageObjects/portal/home_page.py:
```diff
@@ -54,6 +54,11 @@ def go_to_help_and_support_page(self):
         self.browser.find_element_by_id('help_and_support_button').click()
         return help_and_support_page.HelpPage(self.browser)
 
+    def newsletter_singup(self, email):
```
if this isn't used we can delete it :)

mrniket commented on portal/tests/utils/messages.py:
```diff
@@ -81,3 +81,11 @@ def is_class_nonempty_message_showing(browser):
     return is_message_showing(browser, "Your message was sent successfully.")
+
+
+def is_newsletter_signup_successful_message_showing(browser):
```
same here (if it's unused)

CelineBoudier commented:
Reviewed 3 of 8 files at r6.
Review status: all files reviewed at latest revision, 18 unresolved discussions, some commit checks failed.

---

*[portal/context_processors.py, line 41 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LAJ7x_K4O6r5XpJO9nk:-LAJ7x_K4O6r5XpJO9nl:bs6gz5) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/context_processors.py#L41)):*
> ```Python
> def process_newsletter_form(request):
>     return {'news_form': NewsletterForm()
>             }
> ```

new line please

---

*[portal/static/portal/sass/partials/_forms.scss, line 228 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LAJ86ty7_Etxt8zQbgk:-LAJ86ty7_Etxt8zQbgl:b-8m15we) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/static/portal/sass/partials/_forms.scss#L228)):*
> ```SCSS
> 
> .form--news-signup label {
>   color: #ffffff;
> ```

i think we have colours defined in a sass file elsewhere

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702)*
<!-- Sent from Reviewable.io -->

OlafSzmidt commented on portal/tests/utils/messages.py:
```diff
@@ -81,3 +81,11 @@ def is_class_nonempty_message_showing(browser):
 def is_contact_message_sent_message_showing(browser):
     return is_message_showing(browser, "Your message was sent successfully.")
+
+
```
Single line between functions

OlafSzmidt commented:
Review status: 7 of 13 files reviewed at latest revision, 21 unresolved discussions.

---

*[portal/static/portal/sass/partials/_forms.scss, line 228 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LAJ86ty7_Etxt8zQbgk:-LAJ9MeF9u0QlhRwONbL:btq9uwm) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/static/portal/sass/partials/_forms.scss#L228)):*
<details><summary><i>Previously, CelineBoudier (Celine Boudier) wrote…</i></summary><blockquote>

i think we have colours defined in a sass file elsewhere
</blockquote></details>

Colour variables go into: `~/portal/static/portal/sass/modules/_colours.scss` I believe

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702)*
<!-- Sent from Reviewable.io -->

CelineBoudier commented:
Reviewed 1 of 8 files at r6.
Review status: 7 of 13 files reviewed at latest revision, 21 unresolved discussions, some commit checks failed.

---

*[portal/templates/portal/base.html, line 177 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-L9y_aGvw0YNv8kVjO1Z-r6-177:-LAJ95IGDrlppB4s4i-a:b32v57) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/templates/portal/base.html#L177)):*
<details><summary><i>Previously, OlafSzmidt (Olaf Szmidt) wrote…</i></summary><blockquote>

+1
</blockquote></details>

+1

---

*[portal/templates/portal/base.html, line 196 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LAJ9DCD8aYf7FgsK6QY:-LAJ9DCD8aYf7FgsK6QZ:b-6z779x) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/templates/portal/base.html#L196)):*
> ```HTML
>         <li class="error">You appear to be using an old version of Internet Explorer. You will be unable to play Rapid Router, unless you upgrade to Internet Explorer 10, or consider using another browser. See <a href="{% url 'help' %}">Help and Support</a> for more details.</li>
>     </ul>
>     <![endif]-->
> ```

we should remove commented code

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702)*
<!-- Sent from Reviewable.io -->

riaJha97 commented:
Review status: 4 of 12 files reviewed at latest revision, 22 unresolved discussions.

---

*[a discussion](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LAEkcNN7IixVHvf0yze:-LAJCkGB4cGMhKbxvOpU:bz6kwma) (no related file):*
<details><summary><i>Previously, CelineBoudier (Celine Boudier) wrote…</i></summary><blockquote>

"NoSuchElementException: Message: no such element: Unable to locate element: {"method":"id","selector":"messages"}" as seen in the Travis build
</blockquote></details>

Switched to unit tests

---

*[portal/context_processors.py, line 1 at r2](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-L9z27ob3V_JaWNbPhtk:-LAJCq1i3Xyb1GZeqAMf:bb6un1j) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/f79ad78b342044a6ed247a8a9cf922ec45157ec5/portal/context_processors.py#L1)):*
<details><summary><i>Previously, CelineBoudier (Celine Boudier) wrote…</i></summary><blockquote>

if it's a new file we might want to add the copyright notice here as well
</blockquote></details>

Done

---

*[portal/context_processors.py, line 41 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-L9y_aGvw0YNv8kVjO1e-r6-41:-LAJCr6u84zorC5e1rbI:bk8wyd5) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/context_processors.py#L41)):*
<details><summary><i>Previously, mrniket (Niket Shah) wrote…</i></summary><blockquote>

new line here
</blockquote></details>

Added new line

---

*[portal/forms/newsletter_form.py, line 43 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-L9y_aGvw0YNv8kVjO1f-r6-43:-LAJCwf_AGa38C5epMCR:bk8wyd5) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/forms/newsletter_form.py#L43)):*
<details><summary><i>Previously, mrniket (Niket Shah) wrote…</i></summary><blockquote>

new line here too (Sorry)
</blockquote></details>

Added new line

---

*[portal/static/portal/sass/partials/_footer.scss, line 51 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-L9y_aGvw0YNv8kVjO1X-r6-51:-LAJD4dB2V0lP2DMWk50:b-hfpv0l) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/static/portal/sass/partials/_footer.scss#L51)):*
<details><summary><i>Previously, mrniket (Niket Shah) wrote…</i></summary><blockquote>

there's an extra line here
</blockquote></details>

removed

---

*[portal/static/portal/sass/partials/_forms.scss, line 208 at r2](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-L9zOUga9zpJcOg600fz:-LAJCDK_BtjwMgNY44Gb:b-y1kpj1) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/b2a46607e92948422da38de3f6db174094703503/portal/static/portal/sass/partials/_forms.scss#L208)):*
<details><summary><i>Previously, CelineBoudier (Celine Boudier) wrote…</i></summary><blockquote>

why has this been removed?
</blockquote></details>

It was interfering with the alignment of the text field / social media icons

---

*[portal/static/portal/sass/partials/_forms.scss, line 224 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-L9y_aGvw0YNv8kVjO1Y-r6-224:-LAJD75L8jsmtozKoUKs:b-hfpv0l) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/static/portal/sass/partials/_forms.scss#L224)):*
<details><summary><i>Previously, mrniket (Niket Shah) wrote…</i></summary><blockquote>

extra line here
</blockquote></details>

removed

---

*[portal/static/portal/sass/partials/_forms.scss, line 228 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LAJ86ty7_Etxt8zQbgk:-LAJD8G4EjG6mE2EG04r:bb7f2mf) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/static/portal/sass/partials/_forms.scss#L228)):*
<details><summary><i>Previously, OlafSzmidt (Olaf Szmidt) wrote…</i></summary><blockquote>

Colour variables go into: `~/portal/static/portal/sass/modules/_colours.scss` I believe
</blockquote></details>

done

---

*[portal/static/portal/sass/partials/_forms.scss, line 231 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-L9y_aGvw0YNv8kVjO1Y-r6-231:-LAJD9tuDSwCIc1fFhp6:b-hfpv0l) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/static/portal/sass/partials/_forms.scss#L231)):*
<details><summary><i>Previously, mrniket (Niket Shah) wrote…</i></summary><blockquote>

extra line here
</blockquote></details>

removed

---

*[portal/static/portal/sass/partials/_forms.scss, line 237 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-L9y_aGvw0YNv8kVjO1Y-r6-237:-LAJDD5y87BL7py0UF71:bb7f2mf) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/static/portal/sass/partials/_forms.scss#L237)):*
<details><summary><i>Previously, mrniket (Niket Shah) wrote…</i></summary><blockquote>

new line here
</blockquote></details>

done

---

*[portal/templates/portal/base.html, line 230 at r3](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LADS_Tc7_iWrwwk8TFV:-LAJDO9PEBu4iY7pwRJb:b-qybwzr) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/34619bdeeff2c0dbc8117e1249fa46f2d626e0e8/portal/templates/portal/base.html#L230)):*
<details><summary><i>Previously, OlafSzmidt (Olaf Szmidt) wrote…</i></summary><blockquote>

Space between var and curly brackets
</blockquote></details>

It's in the same format in other forms

---

*[portal/templates/portal/base.html, line 177 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-L9y_aGvw0YNv8kVjO1Z-r6-177:-LAJDEaH09aciwP7yley:blu0g5i) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/templates/portal/base.html#L177)):*
<details><summary><i>Previously, CelineBoudier (Celine Boudier) wrote…</i></summary><blockquote>

+1
</blockquote></details>

Celine and James said this was fine

---

*[portal/templates/portal/base.html, line 196 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LAJ9DCD8aYf7FgsK6QY:-LAJDMEmA9m0XOeVM5zx:bgodw0x) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/templates/portal/base.html#L196)):*
<details><summary><i>Previously, CelineBoudier (Celine Boudier) wrote…</i></summary><blockquote>

we should remove commented code
</blockquote></details>

Celine said this was fine

---

*[portal/templates/portal/base.html, line 233 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-L9y_aGvw0YNv8kVjO1Z-r6-233:-LAJE4Da2E7QOQBGpZYZ:beb0wki) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/templates/portal/base.html#L233)):*
<details><summary><i>Previously, mrniket (Niket Shah) wrote…</i></summary><blockquote>

It might be worth asking @j4mesholland where this should go
</blockquote></details>

moved the non_field_errors above the div

---

*[portal/tests/test_newsletter_footer.py, line 14 at r5](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LAEkoPK8cfxhhhqh2uz:-LAJE-8K7aE-1kohqpT3:b7f50hl) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/0e5a30402a4402155ae740f66122533be1132157/portal/tests/test_newsletter_footer.py#L14)):*
<details><summary><i>Previously, CelineBoudier (Celine Boudier) wrote…</i></summary><blockquote>

prints like this should not be committed (useful for testing locally i guess)
</blockquote></details>

removed comments

---

*[portal/tests/test_newsletter_footer.py, line 48 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LAJ1z_mAV0lI6VpW4FX:-LAJE8bjDVTYu89E6BCC:bmsprgj) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/tests/test_newsletter_footer.py#L48)):*
<details><summary><i>Previously, OlafSzmidt (Olaf Szmidt) wrote…</i></summary><blockquote>

> ```
> resp
> ```

Let's call this response instead.
</blockquote></details>

no sorry

---

*[portal/tests/test_newsletter_footer.py, line 56 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LAJ21aqCPeoT5xIPCdn:-LAJEAOm7ZMOZAd-6oFC:b32wsm) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/tests/test_newsletter_footer.py#L56)):*
<details><summary><i>Previously, OlafSzmidt (Olaf Szmidt) wrote…</i></summary><blockquote>

> ```
> resp
> ```

Same here.
</blockquote></details>

no

---

*[portal/views/home.py, line 410 at r3](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LADSdzyDuCm4j7D0SvH:-LAJEHZW9wh37rSEgdTB:blu6880) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/34619bdeeff2c0dbc8117e1249fa46f2d626e0e8/portal/views/home.py#L410)):*
<details><summary><i>Previously, CelineBoudier (Celine Boudier) wrote…</i></summary><blockquote>

+1 - if message.error gives a warning instead, then it's an issue we need to fix
</blockquote></details>

Celine and James said this was okay

---

*[portal/views/home.py, line 92 at r5](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LAIDon-EOFefDqAt0of:-LAJEG4k6jsDrrdFRay_:bb7f2mf) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/0e5a30402a4402155ae740f66122533be1132157/portal/views/home.py#L92)):*
<details><summary><i>Previously, CelineBoudier (Celine Boudier) wrote…</i></summary><blockquote>

there are two spaces between def and render_login_form    There should be only one
</blockquote></details>

done

---

*[portal/tests/pageObjects/portal/home_page.py, line 58 at r2](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-L9zOhqp4TPUgg-TPDIT:-LAJEDKp0MH3JZbhZbep:bm29wqq) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/f79ad78b342044a6ed247a8a9cf922ec45157ec5/portal/tests/pageObjects/portal/home_page.py#L58)):*
<details><summary><i>Previously, CelineBoudier (Celine Boudier) wrote…</i></summary><blockquote>

as said earlier, this will not work for your element 
</blockquote></details>

switched to unit tests

---

*[portal/tests/pageObjects/portal/home_page.py, line 57 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-L9y_aGvw0YNv8kVjO1_-r6-57:-LAJEBWa9s9ZBKfSevJk:b-9unw5o) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/tests/pageObjects/portal/home_page.py#L57)):*
<details><summary><i>Previously, mrniket (Niket Shah) wrote…</i></summary><blockquote>

if this isn't used we can delete it :)
</blockquote></details>

deleted

---

*[portal/tests/utils/messages.py, line 85 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-L9y_aGvw0YNv8kVjO1a-r6-85:-LAJELGL2syjAl_MwycG:bbu7gv1) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/tests/utils/messages.py#L85)):*
<details><summary><i>Previously, OlafSzmidt (Olaf Szmidt) wrote…</i></summary><blockquote>

Single line between functions
</blockquote></details>

function removed

---

*[portal/tests/utils/messages.py, line 86 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-L9y_aGvw0YNv8kVjO1a-r6-86:-LAJEKIA4-rlY5eiFNHc:b-hfpv0l) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/tests/utils/messages.py#L86)):*
<details><summary><i>Previously, mrniket (Niket Shah) wrote…</i></summary><blockquote>

same here (if it's unused)
</blockquote></details>

removed

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702)*
<!-- Sent from Reviewable.io -->

mrniket commented:
Review status: 4 of 12 files reviewed at latest revision, 13 unresolved discussions, some commit checks failed.

---

*[portal/templates/portal/base.html, line 230 at r3](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LADS_Tc7_iWrwwk8TFV:-LAMwh0m819v6aB0N6L8:b-wc34og) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/34619bdeeff2c0dbc8117e1249fa46f2d626e0e8/portal/templates/portal/base.html#L230)):*
<details><summary><i>Previously, riaJha97 wrote…</i></summary><blockquote>

It's in the same format in other forms
</blockquote></details>

Yeah I agreed the current formatting is correct

---

*[portal/templates/portal/base.html, line 177 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-L9y_aGvw0YNv8kVjO1Z-r6-177:-LAMwJED0blQ7j0Og65v:b-ute887) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/templates/portal/base.html#L177)):*
<details><summary><i>Previously, riaJha97 wrote…</i></summary><blockquote>

Celine and James said this was fine
</blockquote></details>

shall we change 'warning' to 'error'?

---

*[portal/tests/test_newsletter_footer.py, line 48 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LAJ1z_mAV0lI6VpW4FX:-LAMwwyH1l4OwJFV1-KH:b-d30r04) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/tests/test_newsletter_footer.py#L48)):*
<details><summary><i>Previously, riaJha97 wrote…</i></summary><blockquote>

no sorry
</blockquote></details>

While I'm not too bothered by this, I do agree that it is more readable to use the full word so that the code reads more like English (which imo makes it easier to understand)

---

*[portal/tests/test_newsletter_footer.py, line 56 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LAJ21aqCPeoT5xIPCdn:-LAMyW0471OWK1wTwi4e:b-d30r04) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/tests/test_newsletter_footer.py#L56)):*
<details><summary><i>Previously, riaJha97 wrote…</i></summary><blockquote>

no
</blockquote></details>

While I'm not too bothered by this, I do agree that it is more readable to use the full word so that the code reads more like English (which imo makes it easier to understand)

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702)*
<!-- Sent from Reviewable.io -->

riaJha97 commented:
Review status: 4 of 12 files reviewed at latest revision, 13 unresolved discussions.

---

*[portal/templates/portal/base.html, line 177 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-L9y_aGvw0YNv8kVjO1Z-r6-177:-LAMzjIy58hlqcHI7e6G:bb6un1j) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/templates/portal/base.html#L177)):*
<details><summary><i>Previously, mrniket (Niket Shah) wrote…</i></summary><blockquote>

shall we change 'warning' to 'error'?
</blockquote></details>

Done

---

*[portal/tests/test_newsletter_footer.py, line 56 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LAJ21aqCPeoT5xIPCdn:-LAN-czt6J_ZM0jxpqyH:b-t45yrv) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/tests/test_newsletter_footer.py#L56)):*
<details><summary><i>Previously, mrniket (Niket Shah) wrote…</i></summary><blockquote>

While I'm not too bothered by this, I do agree that it is more readable to use the full word so that the code reads more like English (which imo makes it easier to understand)
</blockquote></details>

Ok I've changed it 

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702)*
<!-- Sent from Reviewable.io -->

OlafSzmidt commented:
Review status: 4 of 12 files reviewed at latest revision, 11 unresolved discussions, some commit checks failed.

---

*[portal/templates/portal/base.html, line 196 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LAJ9DCD8aYf7FgsK6QY:-LAN0hVlBQJudKRMACXq:bb6qfd1) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/templates/portal/base.html#L196)):*
<details><summary><i>Previously, riaJha97 wrote…</i></summary><blockquote>

Celine said this was fine
</blockquote></details>

????

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702)*
<!-- Sent from Reviewable.io -->

OlafSzmidt commented:
Review status: 4 of 12 files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

*[portal/templates/portal/base.html, line 196 at r6](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LAJ9DCD8aYf7FgsK6QY:-LAN1e0c9jWoVpySPY26:bhhywgj) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/240db028fb3d036c2af16bb3398630721b8d18ae/portal/templates/portal/base.html#L196)):*
<details><summary><i>Previously, OlafSzmidt (Olaf Szmidt) wrote…</i></summary><blockquote>

????
</blockquote></details>

Oh nvm, The code highlighter on github treats this as a comment but it's only the conditional. Keep it in

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702)*
<!-- Sent from Reviewable.io -->

mrniket commented:
Reviewed 2 of 13 files at r1, 1 of 4 files at r2, 1 of 5 files at r4, 5 of 10 files at r7, 2 of 3 files at r8, 1 of 1 files at r9.
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702)*
<!-- Sent from Reviewable.io -->

CelineBoudier commented:
Reviewed 5 of 10 files at r7, 1 of 1 files at r9.
Review status: 11 of 12 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[example_project/example_project/settings.py, line 89 at r1](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LAN8EOj3uMfHTfk1xDF:-LAN8EOj3uMfHTfk1xDG:b-qitt6b) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/6d868e4b24edca8a6409eca49a66d0160ef050b3/example_project/example_project/settings.py#L89)):*
> ```Python
> ]
> 
> MESSAGE_STORAGE = 'django.contrib.messages.storage.cookie.CookieStorage'
> ```

What are the consequences of removing this?

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702)*
<!-- Sent from Reviewable.io -->

CelineBoudier commented:
Reviewed 1 of 1 files at r10.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[example_project/example_project/settings.py, line 89 at r1](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LAN8EOj3uMfHTfk1xDF:-LANASScEJbjKfVwEG_t:b7nw64h) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/6d868e4b24edca8a6409eca49a66d0160ef050b3/example_project/example_project/settings.py#L89)):*
<details><summary><i>Previously, CelineBoudier (Celine Boudier) wrote…</i></summary><blockquote>

What are the consequences of removing this?
</blockquote></details>

huh, what i was referring to disappeared...

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702)*
<!-- Sent from Reviewable.io -->

CelineBoudier commented:
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

*[portal/context_processors.py, line 38 at r10](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-LANKk1493NISE77SLQw:-LANKk1493NISE77SLQx:bwjwp5c) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/1f28e83e41483284cd8cbf6cf2c4009ab9881707/portal/context_processors.py#L38)):*
> ```Python
> # identified as the original program.
> from portal.forms.newsletter_form import NewsletterForm
> ```

another new line please (PEP8)

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702)*
<!-- Sent from Reviewable.io -->

CelineBoudier commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 2 files at r11.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/ocadotechnology/codeforlife-portal/702#-:-LANQjkB1bq8wH0pxNhd:bnfp4nl)*
<!-- Sent from Reviewable.io -->
